### ISSUE TITLE
Extend Workout Entity by Adding New Fields

### ISSUE THREAD

**Issue by makampos** [makampos/fit-sync#14]
## User Story 

**As a:** User  
**I want:** Create a custom workout.
**Then** Set as a field 'visibility' as public(community) or private.
**So that:** I can search for the workout created previously by filtering for _public_, _private_ (created by my own).

## Tips
- Use ENUM do determine the visibility (Doing that we can take the advantage of having more flexibility to work with different types of visibility e.g. [Community = 1, Pro = 2, Public = 3, Private = 4] etc)

## Acceptance Criteria
- Only private workouts can be modified or deleted by who creates them. (if they are set as private)
- System-Default Workouts can be deleted or modified only by users with admin role. (future task improvement)
- Once users set workouts as public(community) it can't be reverted. (**Because it will be accessible to other users to select them and put it into their workout plans**)

---

## Test Scenarios
*Outline the scenarios that will be tested to ensure this user history is implemented correctly. Include both positive and negative test cases.*

1. **Scenario 1:** *Success.*
   - **Given:** *a custom workout filled in*
   - **When:** *created*
   - **Then:** *the workout should be created with visibility 'private'*

2. **Scenario 2:** *Success.*
   - **Given:** *a custom workout is get by ID or all*
   - **When:** *user filter by visibility*
   - **Then:** *only the workouts with target visibility should be returned*

---

## Definition of Done
*Clearly define what it means for this user history to be considered complete. Include criteria such as coding standards, testing requirements, and documentation.*

- [ ] Code implemented and reviewed
- [ ] Unit tests written and passed
- [ ] Integration tests completed
- [ ] Documentation updated
- [ ] User acceptance criteria met

---

**Comment by igorbusto** [comment]
what need be done ? i understand that be done a test, but what function and i dont understand scneario 1 and 2 ?

**Comment by igorbusto** [comment]
A question, i created the visibility, i need additional the variable in the methods "put" and "get" ?

**Comment by makampos** [comment]
> A question, i created the visibility, i need additional the variable in the methods "put" and "get" ?

Yes, once you add a new property in `WorkoutEntity` you will need also to update the mappers in `FitSync.Application.Extensions` for each `WorkoutViewModel`.

Some extra steps are needed;

1. The new property should be visible when users create workouts; (update `AddWorkoutDto` to include it)
2. The new property should be visible when users update workouts; (update `UpdateWorkoutDto` to include it)
